modules/edge-voting/src/voting.rs

```diff
@@ -248,10 +219,40 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
+	pub fn reveal_unsigned(sender: T::AccountId, vote_id: u64, vote: [u8; 32], secret: Option<[u8; 32]>) -> Result {
+		let mut record = <VoteRecords<T>>::get(vote_id).ok_or("Vote record does not exist")?;
+		ensure!(record.data.stage == VoteStage::Voting, "Vote is not in voting stage");
+		// Check vote is for a valid outcome
+		ensure!(record.outcomes.iter().any(|o| o == &vote), "Vote type must be binary");
+		// TODO: Allow changing of votes
+		ensure!(!record.reveals.iter().any(|c| &c.0 == &sender), "Duplicate votes are not allowed");
+
+		// Ensure voter committed
+		if record.data.is_commit_reveal {
+			ensure!(record.commitments.iter().any(|c| &c.0 == &sender), "Duplicate commits are not allowed");
```

Comment:
I think we may just want to knock out some future changes and have the voter pass in the index of their commitment. Therefore, we don't need to be checking these arrays for values when the collections grow quite large.

Reply:
We may need to be completely sure a voter can't commit twice to remove this though.